app/models/host_status/configuration_status.rb

```diff
@@ -0,0 +1,85 @@
+module HostStatus
+  class ConfigurationStatus < Status
+    delegate :error?, :changes?, :pending?, :to => :calculator
+    delegate(*Report::METRIC, :to => :calculator)
+
+    def out_of_sync?
+      if (host && !host.enabled?) || no_reports?
```

Comment:
There are n+1 query warnings coming from Bullet on these type of lines, as `host` is lazily loaded.  You might possibly want to add an `:includes => [:host]` into build_global_status?  (Or optimise it even further, preventing another reload of Host).